code/datums/status_effects/neutral.dm

```diff
@@ -218,3 +218,34 @@
 /// Remove the image from the lwap user's screen
 /obj/effect/temp_visual/lwap_ping/proc/remove_mind(mob/living/looker)
 	looker.client?.images -= lwap_image
+
+/datum/status_effect/delayed
+	id = "delayed_status_effect"
+	status_type = STATUS_EFFECT_MULTIPLE
+	alert_type = null
+	var/prevent_signal = null
+	var/datum/callback/expire_proc = null
+
+/datum/status_effect/delayed/on_creation(mob/living/new_owner, new_duration, datum/callback/new_expire_proc, new_prevent_signal = null)
+	if(!duration || !new_expire_proc || !istype(new_expire_proc))
+		qdel(src)
+		return
+	duration = new_duration
+	expire_proc = new_expire_proc
+	. = ..()
+	if(new_prevent_signal)
+		RegisterSignal(owner, new_prevent_signal, PROC_REF(prevent_action))
+		prevent_signal = new_prevent_signal
+
+/datum/status_effect/proc/prevent_action()
+	SIGNAL_HANDLER
+	qdel(src)
+
+/datum/status_effect/delayed/on_remove()
+	if(prevent_signal)
+		UnregisterSignal(owner, prevent_signal)
+	. = ..()
```

Comment:
shouldnt this only remove the signal if there arent any other delayed status effects on them with the same `prevent signal`

Reply:
nah it works well with multiple signals registered with the same datum and signal type

Reply:
well i explored signal system and here is what i got
this unregister signal affects only one datum status effect, which registered signal earlier
once we get new status_effect/delayed, we get new datum with new registered signal stored in them
so unregistering signal on one of them will affect only that one
btw this UnregisterSignal shouldnt do much cause we actually deleting datum in this proc so anyway signal will be gone
if you're worried about this, i can remove this string
![image](https://github.com/ParadiseSS13/Paradise/assets/61080616/65f56698-dbd1-4eec-8107-861564841ef8)